modules/filters/mod_deflate.c

```diff
@@ -1313,9 +1314,7 @@ static apr_status_t deflate_in_filter(ap_filter_t *f,
                 break;
             }
 
-            if (APR_BUCKET_IS_FLUSH(bkt)) {
-                apr_bucket *tmp_b;
-
+            if (APR_BUCKET_IS_FLUSH(bkt) && !ctx->done) {
```

Comment:
Wouldn't it be more efficient and saving one `APR_BUCKET_IS_FLUSH(bkt)` if we do the following:

```suggestion
            if (APR_BUCKET_IS_FLUSH(bkt)) {
                if (!ctx->done) {
```
And put line 1319 till 1350 in that block?
Of course this makes my first comment moot and `apr_bucket *tmp_b` could stay where it is.

Reply:
Thanks!  Yes, I was playing around with some other changes and ended up with this... makes much more sense that way.